tests/system/test_setup.py

```diff
@@ -0,0 +1,78 @@
+import shutil
+import os
+import unittest
+
+from elasticsearch import Elasticsearch
+from apmserver import BaseTest
+from beat.beat import INTEGRATION_TESTS
+
+
+class SetupTemplateDefaultTest(BaseTest):
+    """
+    Test setup template subcommand with default option.
+    """
+
+    def get_elasticsearch_url(self):
+        """
+        Returns an elasticsearch.Elasticsearch instance built from the
+        env variables like the integration tests.
+        """
+        return "http://{host}:{port}".format(
+            host=os.getenv("ES_HOST", "localhost"),
+            port=os.getenv("ES_PORT", "9200"),
+        )
+
+    def setUp(self):
+        super(BaseTest, self).setUp()
+
+        self.elasticsearch_url = self.get_elasticsearch_url()
+        self.es = Elasticsearch([self.elasticsearch_url])
+
+        try:
+            self.es.indices.delete_template('apm-*')
+        except:
+            pass
+        assert self.es.indices.exists_template(name='apm-*') == False
+
+        shutil.copy(self.beat_path + "/_meta/beat.yml",
+                    os.path.join(self.working_dir, "apm-server.yml"))
```

Comment:
I think subclassing `test_export.SubCommandTest` would eliminate the need for this.

Reply:
I didn't want to derive from `ServerSetUpBaseTest` as this calls `wait_until_startup()` in the `setup()` command, which again expects to see _Starting apm-server_ in the logs, which won't be there. I did not want to go down the path of overwriting this.

Reply:
`SubCommandTest` already implements `wait_until_startup` to cover that.  I wouldn't insist except that I'm worried about having the config generation in multiple places.

Reply:
You're right! I'll change accordingly.